.github/workflows/build.yml

```diff
@@ -0,0 +1,182 @@
+name: Build FreeOrion and run unit tests
+on:
+  pull_request:
+    types:
+      - opened
+      - edited
+      - reopened
+      - synchronize
+  push:
+    branches:
+      - master
+jobs:
+  build-linux:
+    name: Build FreeOrion on Ubuntu 18.04 (Bionic)
+    runs-on: ubuntu-18.04
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v2
+      - uses: actions/cache@v2
+        id: build-linux-install-cache-id
+        with:
+          path: ${{ runner.temp }}/cache-build-linux-install
+          key: build-linux-install-cache-key-v1.0
+      - uses: airvzxf/cache-anything-new-action@v1.0
+        with:
+          script: 'linux-install-build.sh'
+          is_cached: ${{ steps.build-linux-install-cache-id.outputs.cache-hit }}
+          cache: ${{ runner.temp }}/cache-build-linux-install
+          snapshot: '/'
+          exclude: '/boot /data /dev /mnt /proc /run /sys'
+      - name: Prepare ccache timestamp
+        id: build-linux-ccache-timestamp
+        shell: cmake -P {0}
+        run: |
+          string(TIMESTAMP current_date "%Y-%m-%d-%H;%M;%S" UTC)
+          message("::set-output name=timestamp::${current_date}")
+      - name: Cache files with ccache
+        uses: actions/cache@v1.1.0
+        with:
+          path: ${{ runner.temp }}/ccache
+          key: build-linux-ccache-${{ steps.build-linux-ccache-timestamp.outputs.timestamp }}
+          restore-keys: |
+            build-linux-ccache-
+      - name: Show ccache statistics before build
+        env:
+          CACHE_NAME: linux
+          CCACHE_DIR: ${{ runner.temp }}/ccache
+        run: |
+          ccache --show-stats || true
+      - name: Build
+        env:
+          CACHE_NAME: linux
+          CCACHE_DIR: ${{ runner.temp }}/ccache
+          CMAKE_BUILD_PARALLEL_LEVEL: 2
+        run: |
+          mkdir build
+          cd build
+          cmake -DBUILD_TESTING=ON ..
+          cmake --build .
+      - name: Show ccache statistics after build
+        env:
+          CACHE_NAME: linux
+          CCACHE_DIR: ${{ runner.temp }}/ccache
+        run: |
+          ccache --show-stats || true
+      - name: Test
+        run: |
+          cd build
+          cmake --build . --config Release --target unittest || echo Failure
```

Comment:
@o01eg , failure on the test step does not fail the build. Is it intended?  `33% tests passed, 2 tests failed out of 3`

Reply:
Yes, it is intended. We get some strange failure here sometimes.

Reply:
This not how tests supposed to work. If they do not fail it is the same as not having them, but longer.

Do you know if someone has a plan to fix it? 

Anyway, this is definitely out of the scope of this PR.

Reply:
I cann't reproduce it so maybe someone else could fix it.